WordPress/Classes/Networking/ReaderTopicServiceRemote.m

```diff
@@ -101,6 +109,33 @@ - (void)followTopicNamed:(NSString *)topicName
     }];
 }
 
+- (void)fetchSiteInfoForSiteWithID:(NSNumber *)siteID
+                           success:(void (^)(RemoteReaderSiteInfo *siteInfo))success
+                           failure:(void (^)(NSError *error))failure
+{
+    NSString *path =[NSString stringWithFormat:@"sites/%@", siteID];
+
+    [self.api GET:path parameters:nil success:^(AFHTTPRequestOperation *operation, id responseObject) {
+        if (!success) {
+            return;
+        }
+        NSDictionary *response = (NSDictionary *)responseObject;
```

Comment:
Do we wanna add a type safety check? or it's warranteed that this'll always be a dictionary?